File: wear/src/main/java/net/osdn/gokigen/joggingtimer/stopwatch/IWearableActivityControl.java
package net.osdn.gokigen.joggingtimer.stopwatch;

import androidx.appcompat.app.AppCompatActivity;

/**
 *
 *
 *
 */
public interface IWearableActivityControl
{
    void setup(AppCompatActivity activity, IClickCallback callback, IDatabaseReloadCallback dbCallback);
    void setupDatabase(AppCompatActivity activity, boolean isReset);
    void exitApplication(AppCompatActivity activity);

    void vibrate(int duration);

    IDataEntry getDataEntry();

    boolean getDisplayMode();
    void timerStarted(boolean isStarted);
    void setDisplayMode(boolean displayLapTime);
    void addTimeStamp(long count, long lapTime, long diffTime);
    void clearTimeStamp();
    int getLapTimeCount();

    void setupReferenceData();
}

File: wear/src/main/java/net/osdn/gokigen/joggingtimer/stopwatch/IWearableActivityControl.kt
package net.osdn.gokigen.joggingtimer.stopwatch

import androidx.activity.ComponentActivity
import net.osdn.gokigen.joggingtimer.ResultListData

data class LapTimeRecord(val recordType: Int, val recordTime: Long, val recordIndexId: Long)

/**
 *
 */
interface IWearableActivityControl
{
    fun setup(
        activity: ComponentActivity,
        dbCallback: IDatabaseReloadCallback
    )

    fun setupDatabase(activity: ComponentActivity, isReset: Boolean)
    fun exitApplication(activity: ComponentActivity)
    fun vibrate(duration: Int)
    fun getDataEntry(): IDataEntry
    fun getDisplayMode(): Boolean
    fun setDisplayMode(displayLapTime: Boolean)
    fun timerStarted(isStarted: Boolean)
    fun getReferenceTimerSelection(): Int
    fun setReferenceTimerSelection(id: Int)
    fun setupReferenceData()
    fun getCounterRecordList(): List<ResultListData>
    fun isEditableRecord(id: Int): Boolean
    fun getRecordItem(id: Int): ResultListData
    fun getLapTimeList(id: Int): List<LapTimeRecord>
    fun deleteRecord(id: Int)
    fun updateIndexRecord(id: Int, title: String, iconId: Int)
    fun updateRecord(id: Int, title: String, iconId: Int)
    fun setReferenceIndexData(id: Int, iconId: Int)
    fun setReferenceIconId(id: Int, iconId: Int)
    fun launchNotify(isShow: Boolean)
    fun updateTimeEntryData(detailId: Long, totalTime: Long) : Boolean
    fun createTimeEntryModelData(lap: Int, totalLapTime: Long, memo: String): Long
}
